lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp

```diff
@@ -344,4 +354,349 @@ TEST_F(AMFAppProcedureTest, TestPDUSessionSetup) {
+}
+
+TEST_F(AMFAppProcedureTest, TestPDUSession_missing_dnn) {
+  int rc                        = RETURNerror;
+  amf_ue_ngap_id_t ue_id        = 0;
+  AMFClientServicer& clientstub = AMFClientServicer::getInstance();
+  clientstub.msgtype_stack.clear();
+
+  /* Send the initial UE message */
+  imsi64_t imsi64 = 0;
+  imsi64          = send_initial_ue_message_no_tmsi(
+      amf_app_desc_p, 36, 1, 1, 0, plmn, initial_ue_message_hexbuf,
+      sizeof(initial_ue_message_hexbuf));
+
+  /* Check if UE Context is created with correct imsi */
+  EXPECT_TRUE(get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id));
+
+  /* Send the authentication response message from subscriberdb */
+  rc = send_proc_authentication_info_answer(imsi, ue_id, true);
+  EXPECT_TRUE(rc == RETURNok);
+
+  /* Send uplink nas message for auth response from UE */
+  rc = send_uplink_nas_message_ue_auth_response(
+      amf_app_desc_p, ue_id, plmn, ue_auth_response_hexbuf,
+      sizeof(ue_auth_response_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  /* Send uplink nas message for security mode complete response from UE */
+  rc = send_uplink_nas_message_ue_smc_response(
+      amf_app_desc_p, ue_id, plmn, ue_smc_response_hexbuf,
+      sizeof(ue_smc_response_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  send_initial_context_response(amf_app_desc_p, ue_id);
+
+  /* Send uplink nas message for registration complete response from UE */
+  rc = send_uplink_nas_registration_complete(
+      amf_app_desc_p, ue_id, plmn, ue_registration_complete_hexbuf,
+      sizeof(ue_registration_complete_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  ue_m5gmm_context_t* ue_context_p =
+      amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+  // ue context should be exist
+  ASSERT_NE(ue_context_p, nullptr);
+  memset(
+      &ue_context_p->amf_context.apn_config_profile, 0,
+      sizeof(ue_context_p->amf_context.apn_config_profile));
+
+  /* Send uplink nas message for pdu session establishment request from UE */
+  rc = send_uplink_nas_pdu_session_establishment_request(
+      amf_app_desc_p, ue_id, plmn, ue_pdu_session_est_req_missing_dnn_hexbuf,
+      sizeof(ue_pdu_session_est_req_missing_dnn_hexbuf));
+  EXPECT_EQ(rc, RETURNok);
+
+  ue_context_p = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+  // ue context should be exist
+  ASSERT_NE(ue_context_p, nullptr);
+  // smf context should be present
+  EXPECT_EQ(ue_context_p->amf_context.smf_ctxt_map.size(), 0);
+
+  /* Send uplink nas message for deregistration complete response from UE */
+  rc = send_uplink_nas_ue_deregistration_request(
+      amf_app_desc_p, ue_id, plmn, ue_initiated_dereg_hexbuf,
+      sizeof(ue_initiated_dereg_hexbuf));
+
+  EXPECT_TRUE(rc == RETURNok);
+  EXPECT_TRUE(
+      clientstub.msgtype_stack.back() == NGAP_UE_CONTEXT_RELEASE_COMMAND);
+  clientstub.msgtype_stack.pop_back();
+  EXPECT_TRUE(clientstub.msgtype_stack.back() == NGAP_NAS_DL_DATA_REQ);
+}
```

Comment:
addressed review comment, I have predefined the expected MsgId list sent to NGAP for each test case, added a verification check. It will ensure the expected flow